projects/bitcoin-core/build.sh

```diff
@@ -59,21 +59,28 @@ export CPPFLAGS="-D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_DEBUG -DBOOST_M
 
 # Build the fuzz targets
 
-sed -i "s|PROVIDE_FUZZ_MAIN_FUNCTION|NEVER_PROVIDE_MAIN_FOR_OSS_FUZZ|g" "./configure.ac"
-./autogen.sh
+sed -i "s|PROVIDE_FUZZ_MAIN_FUNCTION|NEVER_PROVIDE_MAIN_FOR_OSS_FUZZ|g" "./src/test/fuzz/util/CMakeLists.txt"
 
 # OSS-Fuzz will provide CC, CXX, etc. So only set:
-# * --enable-fuzz, see https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md
-# * CONFIG_SITE, see https://github.com/bitcoin/bitcoin/blob/master/depends/README.md
+# * -DBUILD_FOR_FUZZING=ON, see https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md
+# * --toolchain, see https://github.com/bitcoin/bitcoin/blob/master/depends/README.md
+EXTRA_BUILD_OPTIONS=
 if [ "$SANITIZER" = "memory" ]; then
   # _FORTIFY_SOURCE is not compatible with MSAN.
-  export CPPFLAGS="${CPPFLAGS} -U_FORTIFY_SOURCE"
+  EXTRA_BUILD_OPTIONS="-DAPPEND_CPPFLAGS='-U_FORTIFY_SOURCE'"
 fi
 
-CONFIG_SITE="$PWD/depends/$BUILD_TRIPLET/share/config.site" ./configure --enable-fuzz SANITIZER_LDFLAGS="$LIB_FUZZING_ENGINE"
-make -j$(nproc)
+cmake -B build_fuzz \
+  --toolchain depends/${BUILD_TRIPLET}/toolchain.cmake \
+  -DCMAKE_C_FLAGS_RELWITHDEBINFO="" \
+  -DCMAKE_CXX_FLAGS_RELWITHDEBINFO="" \
```

Comment:
Why is it necessary to set these?

Reply:
CMake provides common language flags, such as `CMAKE_CXX_FLAGS`, as well as build configuration-specific flags, such as `CMAKE_CXX_FLAGS_<CONFIG>`. For Bitcoin Core, the default build configuration is "RelWithDebInfo". Therefore, the `CMAKE_{C,CXX}_FLAGS_RELWITHDEBINFO` flags are applied. Setting these flags to an empty string ensures that the flags set by an OSS-Fuzz environment remain unaltered.

An alternative approach is to set a non-existent build configuration, such as "None".

Please also refer to the previous discussion in https://github.com/hebasto/oss-fuzz/pull/8#discussion_r1679196513.

Reply:
Thanks! Could we add this as a comment here? Specifically, "Setting these flags to an empty string ensures that the flags set by an OSS-Fuzz environment remain unaltered" seems like an important note.

Reply:
> Thanks! Could we add this as a comment here? Specifically, "Setting these flags to an empty string ensures that the flags set by an OSS-Fuzz environment remain unaltered" seems like an important note.

Added.